Migration to v14 fails during run-migrations

Upgrading from nx 13.1.0 to 14.0.5 fails during run migrations

## Current Behavior
Upgrading from nx 13.1.0 to 14.0.5 fails during run migrations with error;
¨
```
Cannot find module '@nrwl/devkit'
```

## Expected Behavior
Migrations should run without problems!

## Steps to Reproduce
N/A

### Failure Logs
nx migrate lates:
```
PS webapp> yarn nx migrate latest
yarn run v1.22.17
$ webapp\node_modules\.bin\nx migrate latest
√ We noticed you are migrating to a new major version, but are not taking advantage of Nx Cloud. Nx Cloud can make your CI up to 10 times faster. Learn more about it here: nx.app. Would you like to add it? · No

Fetching meta data about packages.
It may take a few minutes.
Fetching nx@14.0.5
Fetching nx@latest
Fetching @nrwl/cypress@14.0.5
Fetching @nrwl/jest@14.0.5
Fetching @nrwl/linter@14.0.5
Fetching @nrwl/workspace@14.0.5
Fetching @nrwl/storybook@14.0.5
Fetching @nrwl/react@14.0.5
Fetching @nrwl/web@14.0.5
Fetching @nrwl/cypress@14.0.5
Fetching @nrwl/jest@14.0.5
Fetching @nrwl/linter@14.0.5
Fetching @nrwl/workspace@14.0.5
Fetching @nrwl/storybook@14.0.5
Fetching @nrwl/web@14.0.5
Fetching @nrwl/react@14.0.5

>  NX  The migrate command has run successfully.

- package.json has been updated
- migrations.json has been generated

>  NX  Next steps:

- Make sure package.json changes make sense and then run 'yarn'
- Run 'yarn nx migrate --run-migrations'
- To learn more go to https://nx.dev/using-nx/updating-nx
- You may run 'yarn nx connect-to-nx-cloud' to get faster builds, GitHub integration, and more. Check out https://nx.app
Done in 119.47s.
```

yarn nx migrate --run-migrations

```
PS webapp> yarn nx migrate --run-migrations
yarn run v1.22.17
$ webapp\node_modules\.bin\nx migrate --run-migrations

>  NX  Running 'yarn' to make sure necessary packages are installed

[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning "@mui/lab > @mui/x-date-pickers@5.0.0-alpha.0" has incorrect peer dependency "react@^17.0.2".
warning " > @mui/styles@5.6.2" has incorrect peer dependency "react@^17.0.0".
warning " > @mui/x-data-grid@4.0.2" has unmet peer dependency "@material-ui/core@^4.12.0 || ^5.0.0-beta.0".
warning " > @mui/x-data-grid@4.0.2" has unmet peer dependency "@material-ui/styles@^4.11.4 || ^5.0.0-beta.0".
warning " > @mui/x-data-grid@4.0.2" has incorrect peer dependency "react@^17.0.0".
warning "@mui/x-data-grid > @material-ui/utils@5.0.0-beta.5" has incorrect peer dependency "react@^17.0.2".
warning " > connected-react-router@6.9.2" has unmet peer dependency "history@^4.7.2".
warning " > connected-react-router@6.9.2" has incorrect peer dependency "react@^16.4.0 || ^17.0.0".
warning " > connected-react-router@6.9.2" has incorrect peer dependency "react-redux@^6.0.0 || ^7.1.0".
warning " > react-beautiful-dnd@13.1.0" has incorrect peer dependency "react@^16.8.5 || ^17.0.0".
warning " > react-beautiful-dnd@13.1.0" has incorrect peer dependency "react-dom@^16.8.5 || ^17.0.0".
warning "react-beautiful-dnd > use-memo-one@1.1.2" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "react-router > mini-create-react-context@0.4.1" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0 || ^17.0.0".
warning " > react-table@7.7.0" has incorrect peer dependency "react@^16.8.3 || ^17.0.0-0".
warning " > react-virtualized-auto-sizer@1.0.6" has incorrect peer dependency "react@^15.3.0 || ^16.0.0-alpha || ^17.0.0".
warning " > react-virtualized-auto-sizer@1.0.6" has incorrect peer dependency "react-dom@^15.3.0 || ^16.0.0-alpha || ^17.0.0".
warning " > @storybook/core-server@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @storybook/core-server@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/manager-webpack4@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/manager-webpack4@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/core-common@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/core-common@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/core-client@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/core-client@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/builder-webpack4@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/builder-webpack4@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/store@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/store@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/preview-web@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/preview-web@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/manager-webpack5 > @storybook/ui@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/manager-webpack5 > @storybook/ui@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/theming@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/theming@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/client-api@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/client-api@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/addon-essentials > @storybook/addons@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/addon-essentials > @storybook/addons@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/addon-essentials > @storybook/api@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/addon-essentials > @storybook/api@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/router@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/router@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/components@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/builder-webpack5 > @storybook/components@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/builder-webpack4 > @storybook/components > react-popper-tooltip@3.1.1" has incorrect peer dependency "react@^16.6.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/builder-webpack4 > @storybook/components > react-popper-tooltip@3.1.1" has incorrect peer dependency "react-dom@^16.6.0 || ^17.0.0".
warning "@storybook/core-server > @storybook/builder-webpack4 > @storybook/components > react-textarea-autosize@8.3.3" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@nrwl/cypress > @cypress/webpack-preprocessor@5.11.1" has unmet peer dependency "@babel/preset-env@^7.0.0".
warning "@nrwl/cypress > @cypress/webpack-preprocessor@5.11.1" has unmet peer dependency "webpack@^4 || ^5".
warning "@storybook/addon-essentials > @storybook/addon-actions > react-inspector@5.1.1" has incorrect peer dependency "react@^16.8.4 || ^17.0.0".
warning "@storybook/addon-essentials > @storybook/addon-docs > @mdx-js/react@1.6.22" has incorrect peer dependency "react@^16.13.1 || ^17.0.0".
warning "@storybook/addon-essentials > @storybook/addon-docs > react-element-to-jsx-string@14.3.4" has incorrect peer dependency "react@^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1".
warning "@storybook/addon-essentials > @storybook/addon-docs > react-element-to-jsx-string@14.3.4" has incorrect peer dependency "react-dom@^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1".
warning "@storybook/addon-essentials > @storybook/addon-docs > @storybook/source-loader@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/addon-essentials > @storybook/addon-docs > @storybook/source-loader@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@storybook/react > @storybook/core@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@storybook/react > @storybook/core@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @storybook/builder-webpack5@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @storybook/builder-webpack5@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @storybook/manager-webpack5@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @storybook/manager-webpack5@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @storybook/react@6.4.22" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @storybook/react@6.4.22" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @welldone-software/why-did-you-render@6.2.1" has incorrect peer dependency "react@^16 || ^17".
warning " > babel-loader@8.1.0" has unmet peer dependency "webpack@>=2".
warning "eslint-config-airbnb-typescript > @typescript-eslint/parser@4.33.0" has incorrect peer dependency "eslint@^5.0.0 || ^6.0.0 || ^7.0.0".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb@18.2.1" has incorrect peer dependency "eslint@^5.16.0 || ^6.8.0 || ^7.2.0".
warning "eslint-config-airbnb-typescript > eslint-config-airbnb-base@14.2.1" has incorrect peer dependency "eslint@^5.16.0 || ^6.8.0 || ^7.2.0".
warning " > eslint-config-react-app@6.0.0" has incorrect peer dependency "@typescript-eslint/eslint-plugin@^4.0.0".
warning " > eslint-config-react-app@6.0.0" has incorrect peer dependency "@typescript-eslint/parser@^4.0.0".
warning " > eslint-config-react-app@6.0.0" has unmet peer dependency "babel-eslint@^10.0.0".
warning " > eslint-config-react-app@6.0.0" has incorrect peer dependency "eslint@^7.5.0".
warning " > eslint-config-react-app@6.0.0" has unmet peer dependency "eslint-plugin-flowtype@^5.2.0".
warning " > eslint-plugin-jsx-a11y@6.4.1" has incorrect peer dependency "eslint@^3 || ^4 || ^5 || ^6 || ^7".
warning " > eslint-webpack-plugin@2.6.0" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning " > url-loader@3.0.0" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
[4/4] Building fresh packages...
success Saved lockfile.

>  NX  Running migrations from 'migrations.json'

Running migration set-parallel-default
Cannot find module '@nrwl/devkit'
Require stack:
- webapp\node_modules\@nrwl\workspace\src\migrations\update-13-2-0\set-parallel-default.js
- C:\Users\JAKOBS~1.NOV\AppData\Local\Temp\tmp-26792-ZA3K86hgvJVb\node_modules\nx\src\command-line\migrate.js
- C:\Users\JAKOBS~1.NOV\AppData\Local\Temp\tmp-26792-ZA3K86hgvJVb\node_modules\nx\src\command-line\nx-commands.js
- C:\Users\JAKOBS~1.NOV\AppData\Local\Temp\tmp-26792-ZA3K86hgvJVb\node_modules\nx\bin\init-local.js
- C:\Users\JAKOBS~1.NOV\AppData\Local\Temp\tmp-26792-ZA3K86hgvJVb\node_modules\nx\bin\nx.js
- C:\Users\JAKOBS~1.NOV\AppData\Local\Temp\tmp-26792-ZA3K86hgvJVb\node_modules\@nrwl\tao\index.js
Done in 62.69s.
```

### Environment
win 11

--- a/packages/workspace/package.json
+++ b/packages/workspace/package.json
@@ -64,6 +64,7 @@
   },
   "dependencies": {
     "@parcel/watcher": "2.0.4",
+    "@nrwl/devkit": "*",
     "chokidar": "^3.5.1",
     "cli-cursor": "3.1.0",
     "cli-spinners": "2.6.1",

--- a/scripts/depcheck/missing.ts
+++ b/scripts/depcheck/missing.ts
@@ -119,7 +119,6 @@ const IGNORE_MATCHES = {
     '@nrwl/cli',
     '@nrwl/jest',
     '@nrwl/linter',
-    '@nrwl/devkit',
     'tsconfig-paths',
     '@swc-node/register',
   ],

